### PR TITLE
fix: forest binary reorg

### PR DESCRIPTION
**Pull request by elmattic**
**Summary of changes**
Changes introduced in this pull request:
- This PR is a following PR reflecting the changes in forest repository
  - https://github.com/ChainSafe/forest/pull/3355
  - https://github.com/ChainSafe/forest/pull/3336
  - https://github.com/ChainSafe/forest/pull/3362

Note that some PR have been already merged.

**Reference issue to close (if applicable)**

Closes 

**Other information and links**

### CODE REVIEW

**Review by jdjaustin — APPROVED** [review]

**Review by hanabi1224 — APPROVED** [review]

**Comment by elmattic** [comment]
@LesnyRumcajs Can you check the TF part as well?

File: terraform/modules/filecoin_node/forest.sh [hunk 1]
```diff
@@ -52,7 +52,7 @@ sudo --user="${NEW_USER}" -- \
   --network=forest \
   --name=forest-"${CHAIN}" \
   --volume=/home/"${NEW_USER}"/forest_data:/home/"${NEW_USER}"/forest_data:z \
-  --publish=1234:1234 \
```
[the remark this answers is not in the document]
Comment (LesnyRumcajs) [reply terraform/modules/filecoin_node/forest.sh:55]: The ports _can_ be changed, they just need to cover both 1234 and 2345. Unfortunately, other commands cannot.

File: terraform/modules/sync_check/service/sync_check.rb [hunk 2]
```diff
@@ -51,8 +51,8 @@ def get_and_assert_env_variable(name)
   client.post_message "✅ Sync check for #{hostname} passed. 🌲🌳🌲🌳🌲"
 else
   client.post_message "⛔ Sync check for #{hostname} fiascoed. 🔥🌲🔥"
-  SyncCheck.new.run_forest_cli("--chain #{network} db clean --force")
-  logger.info 'DB Cleaned'
+  SyncCheck.new.run_forest_cli("db destroy --chain #{network} --force")
```
Comment (LesnyRumcajs) [inline terraform/modules/sync_check/service/sync_check.rb:54]: It would be best to have it as a separate PR. Sync check runs on the `edge` tag, which will get pushed as soon as the re-org PR is merged in Forest.

Comment (LesnyRumcajs) [inline terraform/modules/sync_check/service/sync_check.rb:54]: What command will it invoke in the end? Are you it's what you want?

**Review by LesnyRumcajs — DISMISSED** [review]
1. Port changes can be a separate PR. We can expose 1234 and 2345 so they work with binaries before and after the default port change.
2. That said, it won't give much as long as the commands are not backwards compatible, i.e., if there's no `forest-tool db destroy` in the current version, it will just fail miserably.

We need to either:
- ensure both versions (current one and unreleased containing reorg changes) work fine - this requires having logic for handling this, either in Forest (e.g., `forest-cli db clean` would forward the call to `forest-tool db destroy`) or in here (check Forest version and use correct set of commands based on that. This may be an overkill.
- Merge the changes from this PR only after a version has been released. This approach invalidates exposing both 1234 and 2345, as there would be only one "true" version.

Sync check changes would need to be in a separate PR and follow the same logic - either merge as soon as reorg is merged in Forest or have a compatibility layer.

File: terraform/modules/sync_check/service/sync_check.rb (hunk 2 above)
Comment (elmattic) [reply terraform/modules/sync_check/service/sync_check.rb:54]: It will do the same than before, the subcommand `forest-cli clean` was rename to `forest-tool destroy`.

Comment (LesnyRumcajs) [reply terraform/modules/sync_check/service/sync_check.rb:54]: is it `run_forest_cli` or `run_forest_tool`? :)

**Comment by elmattic** [comment]
> @elmattic have you tested these changes making sure they are working as intended ?

How exactly ? Do we have some documentation for this ?

**Comment by LesnyRumcajs** [comment]
> > > @elmattic have you tested these changes making sure they are working as intended ?
> > 
> > 
> > How exactly ? Do we have some documentation for this ?
> 
> Yes, you're able to manually execute test samples for various services. The steps for each can be found within their respective folders in the Terraform directory.

That's a bit misleading. The steps there are **not** intended for manual testing and should **not** be performed on branches. Otherwise, all the prod services will get terraformed. Would be great to have a more detailed instructions on testing for future reference.

File: terraform/modules/sync_check/service/sync_check.rb (hunk 2 above)
Comment (elmattic) [reply terraform/modules/sync_check/service/sync_check.rb]: I will keep this PR with the updated Sync check. Should I remove my modification to the snapshot upload as well?

File: terraform/modules/filecoin_node/forest.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (elmattic) [reply terraform/modules/filecoin_node/forest.sh]: @LesnyRumcajs Created a dedicated PR for port change: https://github.com/ChainSafe/forest-iac/pull/221

File: terraform/modules/filecoin_node/firewall.tf [hunk 3]
```diff
@@ -9,7 +9,7 @@ resource "digitalocean_firewall" "forest_firewall" {
 
   inbound_rule {
     protocol         = "tcp"
-    port_range       = "1234"
+    port_range       = "2345"
     source_addresses = var.source_addresses
   }
```
[the remark this answers is not in the document]
Comment (elmattic) [reply terraform/modules/filecoin_node/firewall.tf:15]: Oh ok, sure. Thought it was forest specific because of the `"forest_firewall"`.
